src/mongoose_client_api/mongoose_client_api_rooms.erl

```diff
@@ -111,20 +111,29 @@ from_json(Req, State) ->
         {ok, #{<<"name">> := N, <<"subject">> := S} = JSONData} when is_binary(N), is_binary(S) ->
             handle_request(Method, JSONData, Req2, State);
         _ ->
-            {false, Req, State}
+            mongoose_client_api:bad_request(Req, <<"Failed to parse parameters">>, State)
     end.
 
 handle_request(Method, JSONData, Req, State) ->
     case handle_request_by_method(Method, JSONData, Req, State) of
-        {error, _, _} ->
-            {false, Req, State};
+        {error, Short, Desc} = Other ->
+            mongoose_client_api:bad_request(Req, format_error(Short, Desc), State);
```

Comment:
Can we have all this "better reporting" in a separate PR? This one is sufficiently enormous, and rest reporting errors is a different topic. There's also the question of whether we want to log the entire requests as I mentioned before, which I doubt we want. So these things could be moved to a different PR where we discuss such thing without interrupting mam's multitenancy.